src/main/java/TicTacToe.java

```
@@ -0,0 +1,327 @@
+                continue ;
+            }
+        }
+        return move_int ;
+    }
+
+}
+
+class TicTacToe
+{
+    protected static final int N = 3 ;
+    private static final int HSPACE = 20 ;
+    protected int[][] board;
+    private static BufferedReader reader =
+        new BufferedReader(new InputStreamReader(System.in)) ;
+
+    class Player
+    {
+        private String name ;
+        private int player_type ;
+        private int player_order ;
+        private MoveMethod move_strategy ;
+
+        public Player(String pname, int type, int order, MoveMethod move_s )
+        {
+            name = pname ;
+            player_type = type ;
+            player_order = order ;
+            move_strategy = move_s ;
+        }
+
+        public String getName() {
+            return name ;
+        }
+
+        public int getPlayerType() {
+            return player_type ;
+        }
+
+        public int getMove() {
+            return move_strategy.move();
+        }
+    }
+
+    private Player player1,player2;
+
+    public Player getplayer1() {
+        return player1 ;
+    }
+
+    public Player getplayer2() {
+        return player2 ;
+    }
+
+    public static String getPosDescription(int pos) {
+        String str = "";
+        if( pos == 5 ) {
+            str = "center" ;
+            return str ;
+        }
+
+        if( (pos-1)/3 == 0 ) {
+            str += "upper " ;
+        }
+        else if( (pos-1)/3 == 1 ) {
+            str += "middle " ;
+        }
+        else
+            str += "lower " ;
+
+        if( (pos-1) % 3 == 0 )
+            str += "left" ;
+        else if( (pos-1)%3 == 1 )
+            str += "middle" ;
+        else
+            str += "right" ;
+
+        return str ;
+    }
+
+    protected static String getUserInput() {
+        String input = "" ;
+        try {
+            input = reader.readLine();
+        } catch (IOException ex) {
+            ex.printStackTrace();

```

The end user probably doesn't want to see a stacktrace or know what to do. 
This should throw a checked exception that can be handled and the end user can be informed nicely.